datafusion/sqllogictest/test_files/string/string.slt

```diff
@@ -41,141 +41,6 @@ select arrow_cast(col1, 'Utf8') as c1 from test_substr_base;
 #
 include ./string_query.slt.part
 
-# TODO support all String types in sql_like_to_expr and move this test to `string_query.slt.part`
```

Comment:
By moving these tests to `string_query.slt.part` they now run for strings, large strings, stringview and dictionary arrays

FYI @goldmedal